doc/devel/documenting_mpl.rst

```diff
@@ -91,270 +87,108 @@ statement, such as::
-
-  .. [#] For example.
-
-* Use the *note* and *warning* directives, sparingly, to draw attention to
-  important comments:
-
-  .. code-block:: rst
-
-    .. note::
-       Here is a note.
-
-    .. warning::
-       Here is a warning.
-
-  yields:
-
-  .. note::
-     Here is a note.
-
-  .. warning::
-     Here is a warning.
-
-* Use the *deprecated* directive when appropriate:
-
-  .. code-block:: rst
+.. code-block:: python
 
-    .. deprecated:: 0.98
-       This feature is obsolete, use something else.
+  def hlines(self, y, xmin, xmax, colors='k', linestyles='solid',
+             label='', **kwargs):
+        """
+        Plot horizontal lines at each `y` from `xmin` to `xmax`.
 
-  yields:
+        Parameters
+        ----------
+        y : scalar or sequence of scalar
+            y-indexes where to plot the lines.
 
-  .. deprecated:: 0.98
-     This feature is obsolete, use something else.
+        xmin, xmax : scalar or 1D array_like
+            Respective beginning and end of each line. If scalars are
+            provided, all lines will have same length.
 
-* Use the *versionadded* and *versionchanged* directives, which have similar
-  syntax to the *deprecated* role:
+        colors : array_like of colors, optional, default: 'k'
 
-  .. code-block:: rst
+        linestyles : ['solid' | 'dashed' | 'dashdot' | 'dotted'], optional
 
-    .. versionadded:: 0.98
-       The transforms have been completely revamped.
+        label : string, optional, default: ''
 
-  .. versionadded:: 0.98
-     The transforms have been completely revamped.
+        Returns
+        -------
+        lines : `~matplotlib.collections.LineCollection`
 
-* Use the *seealso* directive, for example:
+        Other Parameters
+        ----------------
+        **kwargs :  `~matplotlib.collections.LineCollection` properties.
 
-  .. code-block:: rst
+        See also
+        --------
+        vlines : vertical lines
+        axhline: horizontal line across the axes
+        """
 
-    .. seealso::
+The Sphinx website also contains plenty of documentation_ concerning ReST
+markup and working with Sphinx in general.
 
-       Using ReST :ref:`emacs-helpers`:
-          One example
+Function arguments
+------------------
+Function arguments and keywords within docstrings should be referred to using
+the ``*emphasis*`` role. This will keep Matplotlib's documentation consistent
+with Python's documentation:
 
-       A bit about :ref:`referring-to-mpl-docs`:
-          One more
+.. code-block:: rst
 
-  yields:
+  Here is a description of *argument*
 
-  .. seealso::
+Please do not use the ```default role```:
 
-     Using ResT :ref:`emacs-helpers`:
-        One example
 
-     A bit about :ref:`referring-to-mpl-docs`:
-        One more
+.. code-block:: rst
 
-* Please keep the :ref:`glossary` in mind when writing documentation. You can
-  create a references to a term in the glossary with the ``:term:`` role.
+   Please do not describe `argument` like this.
```

Comment:
I think a (short) explanation similar to the one I gave at https://github.com/matplotlib/matplotlib/issues/9786#issuecomment-344458838 (about the meaning of single backticks) may be helpful.